pkg/controller/stackconfigpolicy/controller.go

```diff
@@ -414,98 +530,161 @@ func (r *ReconcileStackConfigPolicy) updateStatus(ctx context.Context, scp polic
+				"owner_namespace", softOwner.Namespace, "owner_name", softOwner.Name)
 
-		log.V(1).Info("Reconcile empty file settings Secret for Elasticsearch",
-			"namespace", esNsn.Namespace, "es_name", esNsn.Name,
-			"owner_namespace", softOwner.Namespace, "owner_name", softOwner.Name)
+			var es esv1.Elasticsearch
+			err := c.Get(ctx, namespacedName, &es)
+			if err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
+			if apierrors.IsNotFound(err) {
+				// Elasticsearch has just been deleted
+				return nil
+			}
 
-		var es esv1.Elasticsearch
-		err := c.Get(ctx, esNsn, &es)
-		if err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-		if apierrors.IsNotFound(err) {
-			// Elasticsearch has just been deleted
-			return nil
+			err = filesettings.ReconcileEmptyFileSettingsSecret(ctx, c, es, false)
+			if err != nil {
+				return err
+			}
+		case kblabel.Type:
+			// Currently we do not reset labels for kibana, so we shouldn't hit this.
+			// Implement if needed in the future
+			continue
+		default:
+			return fmt.Errorf("secret configured for unknown application type %s", configuredApplicationType)
 		}
-
-		return filesettings.ReconcileEmptyFileSettingsSecret(ctx, c, es, false)
 	}
 	return nil
 }
 
-// deleteOrphanSoftOwnedSecrets deletes secrets for the Elasticsearch clusters that are no longer configured
+// deleteOrphanSoftOwnedSecrets deletes secrets for the Elasticsearch/Kibana clusters that are no longer configured
 // by a given StackConfigPolicy.
-func deleteOrphanSoftOwnedSecrets(ctx context.Context, c k8s.Client, softOwner types.NamespacedName, configuredEs esMap) error {
+func deleteOrphanSoftOwnedSecrets(
+	ctx context.Context,
+	c k8s.Client,
+	softOwner types.NamespacedName,
+	configuredESResources esMap,
+	configuredKibanaResources kbMap,
+	resourceType policyv1alpha1.ResourceType,
+) error {
 	var secrets corev1.SecretList
+	matchLabels := client.MatchingLabels{
+		reconciler.SoftOwnerNamespaceLabel:              softOwner.Namespace,
+		reconciler.SoftOwnerNameLabel:                   softOwner.Name,
+		reconciler.SoftOwnerKindLabel:                   policyv1alpha1.Kind,
+		commonlabels.StackConfigPolicyOnDeleteLabelName: commonlabels.OrphanSecretDeleteOnPolicyDelete,
+	}
+
+	if resourceType != "" {
+		matchLabels[commonv1.TypeLabelName] = string(resourceType)
+	}
 	if err := c.List(ctx,
 		&secrets,
 		// search in all namespaces
 		// restrict to secrets on which we set the soft owner labels
-		client.MatchingLabels{
-			reconciler.SoftOwnerNamespaceLabel:              softOwner.Namespace,
-			reconciler.SoftOwnerNameLabel:                   softOwner.Name,
-			reconciler.SoftOwnerKindLabel:                   policyv1alpha1.Kind,
-			commonlabels.StackConfigPolicyOnDeleteLabelName: commonlabels.OrphanObjectDeleteOnPolicyDelete,
-		},
+		matchLabels,
 	); err != nil {
 		return err
 	}
 
 	for i := range secrets.Items {
 		secret := secrets.Items[i]
```

Comment:
I think it is because the linter will complain when we try send the memory address of the secret here
```
err := c.Delete(ctx, &secret)
		if err != nil && !apierrors.IsNotFound(err) {
			return err
		}
```
referencing memory of temporary loop variables